java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,343 @@
+                        } else {
+                            timeMilliVector.setNull(rowCount);
+                        }
+
+                        break;
+                    case Types.TIMESTAMP:
+                        // timezone is null
+                        Timestamp timestamp = rs.getTimestamp(i);
+                        TimeStampVector timeStampVector = (TimeStampVector)root.getVector(columnName);
+                        timeStampVector.setValueCount(rowCount + 1);
+                        if (timestamp != null) {
+                            timeStampVector.setSafe(rowCount, timestamp.getTime());
+                        } else {
+                            timeStampVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.BINARY:
+                    case Types.VARBINARY:
+                    case Types.LONGVARBINARY:
+                        VarBinaryVector varBinaryVector = (VarBinaryVector)root.getVector(columnName);;
+                        varBinaryVector.setValueCount(rowCount + 1);
+                        byte[] bytes = rs.getBytes(i);
+                        if (bytes != null) {
+                            varBinaryVector.setIndexDefined(rowCount);
+                            varBinaryVector.setValueLengthSafe(rowCount, bytes.length);
+                            varBinaryVector.setSafe(rowCount, bytes);
+                        } else {
+                            varBinaryVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.ARRAY:
+                        // not handled
+//                    fields.add(new Field("list", FieldType.nullable(new ArrowType.List()), null));
+                        break;
+                    case Types.CLOB:
+                        VarCharVector varcharVector1 = (VarCharVector)root.getVector(columnName);
+                        varcharVector1.setValueCount(rowCount + 1);
+                        Clob clob = rs.getClob(i);
+                        if (clob != null) {
+                            int length = (int) clob.length();
+                            varcharVector1.setIndexDefined(rowCount);
+                            varcharVector1.setValueLengthSafe(rowCount, length);
+                            varcharVector1.setSafe(rowCount, clob.getSubString(1, length).getBytes(), 0, length);
+                        } else {
+                            varcharVector1.setNull(rowCount);
+                        }
+                        break;
+                    case Types.BLOB:
+                        VarBinaryVector varBinaryVector1 = (VarBinaryVector)root.getVector(columnName);;
+                        varBinaryVector1.setValueCount(rowCount + 1);
+                        Blob blob = rs.getBlob(i);
+                        if (blob != null) {
+                            byte[] data = blob.getBytes(0, (int) blob.length());
```

Comment:
same as clob, a streaming api to access data might be a better choice to not pressure JVM heap too much...